src/ai/backend/agent/stats.py

```diff
@@ -454,3 +475,105 @@ async def _pipe_builder(r: Redis) -> Pipeline:
             return pipe
 
         await redis_helper.execute(self.agent.redis_stat_pool, _pipe_builder)
+
+    async def collect_per_container_process_stat(
+        self,
+        container_ids: Sequence[ContainerId],
+    ) -> None:
+        """
+        Collect the per-container process statistics only,
+
+        Intended to be used by the agent.
+        """
+        async with self._lock:
+            pid_map = {}
+            total_pids = []
+            pids = []
+            for cid in container_ids:
+                try:
+                    docker = aiodocker.Docker()
+                    result = await docker._query_json(f"containers/{cid}/top", method="GET")
+                    procs = result["Processes"]
+                    pids = [PID(int(proc[1])) for proc in procs if proc[0] != "root"]
+                    if os.getuid() == 0:
+                        pids.extend([PID(int(proc[1])) for proc in procs if proc[0] == "root"])
+                    unused_pids = set(self.process_metrics[cid].keys()) - set(pids)
+                except (KeyError, aiodocker.exceptions.DockerError):
+                    log.warning(
+                        "collect_per_container_process_stat(): cannot found container {}", cid
+                    )
+                else:
+                    for unused_pid in unused_pids:
+                        log.debug("removing pid_metric for {}: {}", cid, unused_pid)
+                        self.process_metrics[cid].pop(unused_pid, None)
+                finally:
+                    await docker.close()
+                for pid in pids:
+                    pid_map[pid] = cid
+                total_pids.extend(pids)
```

Comment:
What happens if original process represented by the PID is killed and the PID is reused by another process?